### ISSUE TITLE
A problem about total tranining steps of reader

### ISSUE THREAD

**Issue by SparkJiao** [AkariAsai/learning_to_retrieve_reasoning_paths#20]
Hi, very appreciate to your contribution.

I have a question about the training steps setting here:

[https://github.com/AkariAsai/learning_to_retrieve_reasoning_paths/blob/master/reader/run_reader_confidence.py#L209-L210](https://github.com/AkariAsai/learning_to_retrieve_reasoning_paths/blob/master/reader/run_reader_confidence.py#L209-L210)

```
num_train_optimization_steps = int(
            len(train_examples) / args.train_batch_size / args.gradient_accumulation_steps) * args.num_train_epochs
```

I think the `len(train_examples)` should be replaced with `len(train_features)` since the total length of dataset is the length of all processed features, whose amount is much more than that of the initial examples.

**Comment by AkariAsai** [comment]
Hi @SparkJiao, thank you for raising this issue! 

The `num_train_optimiization_steps`follows the way implemented in the [pytorch-pretrained-bert (v0.6.1) run_squad.py](https://github.com/huggingface/transformers/blob/v0.6.1/examples/run_squad.py#L890). 

As you mentioned, the total length of the examples can be smaller than the total numbers of the original training data as we split the documents longer than the `max_token_length` and add features for each in the reading comprehension model [here](https://github.com/AkariAsai/learning_to_retrieve_reasoning_paths/blob/8bc397afae9d1e4363609a7bc171fe450afa502b/reader/rc_utils.py#L623). 
As the average length of Wikipedia passage is around 100 tokens and we've observed that as long as we set the number of the max input length to a decent number (e.g., 386), the difference between the number of the train examples and the number of the train features is somewhat limited. 

Due to the reasons discussed above, we follow the original example script from the huggingface library and expect that it does not cause a large performance jump. If you find that replacing it with `train_features` would lead to a large performance gap, please contact us!

**Comment by AkariAsai** [comment]
I'm closing this issue now, but if you have any followup questions, please feel free to re-open or contact me via email :)
